### ISSUE TITLE
error warning it site editor with duotones

### ISSUE THREAD

**Issue by tkapler** [WordPress/gutenberg#50200]
### Description

I do have custom duotones in theme.json, they work fine

this is the definition btw

```
            "duotone": [
                {
                    "name": "DT Powerbox 200 to 300",
                    "slug": "dt-powerbox-200-300",
                    "colors": [
                        "var(--wp--preset--color--powerbox-200)",
                        "var(--wp--preset--color--powerbox-300)"
                    ]
                },
                {
                    "name": "DT Powerbox 700 to 800",
                    "slug": "dt-powerbox-700-800",
                    "colors": [
                        "var(--wp--preset--color--powerbox-700)",
                        "var(--wp--preset--color--powerbox-800)"
                    ]
                }
			],
```
			
			
But when i do use gutenberg plugin and having debug=true theese errors popup when I open Site Editor:

<b>Warning</b>:  Trying to access array offset on value of type null in <b>/var/www/vhosts/powerbox.one/httpdocs/wp-content/plugins/gutenberg/lib/block-supports/duotone.php</b> on line <b>344</b><br />
 <b>Warning</b>:  Trying to access array offset on value of type null in <b>/var/www/vhosts/powerbox.one/httpdocs/wp-content/plugins/gutenberg/lib/block-supports/duotone.php</b> on line <b>345</b><br />
  | <b>Warning</b>:  Trying to access array offset on value of type null in <b>/var/www/vhosts/powerbox.one/httpdocs/wp-content/plugins/gutenberg/lib/block-supports/duotone.php</b> on line <b>346</b><br />
  | <b>Warning</b>:  Trying to access array offset on value of type null in <b>/var/www/vhosts/powerbox.one/httpdocs/wp-content/plugins/gutenberg/lib/block-supports/duotone.php</b> on line <b>347</b><br />

### Step-by-step reproduction instructions

debug = true
set up duotones in theme.json
enable gutenberg plugin
go to site editor

### Environment info

latest versions of guenberg and wordpress

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by tkapler** [comment]
Interestingly - when i turn off the gutenberg plugin, the warning in site editor disapears, but instead theese errors appear on frontend

Warning: Trying to access array offset on value of type null in /var/www/vhosts/powerbox.one/httpdocs/wp-includes/block-supports/duotone.php on line 416

Warning: Trying to access array offset on value of type null in /var/www/vhosts/powerbox.one/httpdocs/wp-includes/block-supports/duotone.php on line 417

Warning: Trying to access array offset on value of type null in /var/www/vhosts/powerbox.one/httpdocs/wp-includes/block-supports/duotone.php on line 418

Warning: Trying to access array offset on value of type null in /var/www/vhosts/powerbox.one/httpdocs/wp-includes/block-supports/duotone.php on line 419

**Comment by glendaviesnz** [comment]
I was able to replicate this issue. It looks like currently the `theme.json` duotone settings are [only able to support hex and RGB colors.](https://developer.wordpress.org/themes/advanced-topics/theme-json/#duotone).

With debug=false I was able to add CSS vars, and the options displayed in the block toolbar, but once selected they did not work:

https://user-images.githubusercontent.com/3629020/235404865-d7fda690-f120-44e2-980e-a576db7bd9c3.mp4

We probably should look to [add a guard here](https://github.com/WordPress/gutenberg/blob/trunk/lib/class-wp-duotone-gutenberg.php#L531) to prevent these warnings if the theme.json has invalid color settings.
